internal/file/zip_file_traversal.go

```diff
@@ -91,6 +92,12 @@ func ExtractFromZipToUniqueTempFile(archivePath, dir string, paths ...string) (m
 		// provides a ReadCloser. It is up to the caller to handle closing the file explicitly.
 		defer tempFile.Close()
 
+		// grab and assign digest for the visited zip file
+		digest, err := Digest(file.FileInfo().Name())
```

Comment:
This seems like the wrong spot for this since this is a generic archiver helper, and not necessarily scoped to just java. Maybe moving the digestion to https://github.com/anchore/syft/blob/v0.43.2/syft/pkg/cataloger/java/archive_parser.go#L72-L80 makes more sense?

Reply:
moved into the archive parser